internal/databroker/server.go

```diff
@@ -70,19 +78,19 @@ func New(options ...ServerOption) *Server {
 			srv.mu.RUnlock()
 
 			for _, recordType := range recordTypes {
-				db, err := srv.getDB(recordType)
+				db, _, err := srv.getDB(recordType, true)
 				if err != nil {
 					continue
 				}
-				db.ClearDeleted(context.Background(), time.Now().Add(-cfg.deletePermanentlyAfter))
+				db.ClearDeleted(context.Background(), tm)
 			}
 		}
 	}()
 	return srv
 }
 
 func (srv *Server) initVersion() {
-	dbServerVersion, err := srv.getDB(recordTypeServerVersion)
+	dbServerVersion, _, err := srv.getDB(recordTypeServerVersion, false)
```

Comment:
this is the only place we do not lock when getting db right?

Reply:
We don't need because this's only run once at server startup.